pycargoebuild/cargo.py

```diff
@@ -186,7 +208,8 @@ def get_crates(f: typing.BinaryIO) -> typing.Generator[Crate, None, None]:
                 yield FileCrate(name=p["name"],
                                 version=p["version"],
                                 checksum=p["checksum"])
-            elif p["source"].startswith("git+https://github.com/"):
+            elif (p["source"].startswith("git+https://github.com/") or
+                  p["source"].startswith("git+https://gitlab.")):
```

Comment:
Hmm, let's make go for simple `git+` and have `GitCrate` check the exact URL instead and fail when dealing with unsupported source.